### ISSUE TITLE
pip is missing after update to 10.11.6

### ISSUE THREAD

**Issue by jwedel** [Homebrew/homebrew-core#10436]
# Please follow the general troubleshooting steps first:

- [x] Ran `brew update` and retried your prior step?
- [x] Ran `brew doctor`, fixed as many issues as possible and retried your prior step?
- [x] Confirmed this is problem with specific formulae and not Homebrew/brew? If it's a general Homebrew/brew problem please file this issue at https://github.com/Homebrew/brew/issues/new

### Bug reports:

After I updated my MacOS, and then ran python, some modules were missing. Then I tried to check the status using pip and found out that even pip was not there anymore.
I tried to reinstall python and got an error during postinstall step.

*brew config*
```
HOMEBREW_VERSION: 1.1.10
ORIGIN: https://github.com/Homebrew/brew.git
HEAD: 69d1cedd12d37d6a9535fbba17d2129b165c08f8
Last commit: 13 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: f58aa629a583e8e4c15cff2661c0d2a9f80c3078
Core tap last commit: 2 hours ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_REPOSITORY: /usr/local/Homebrew
HOMEBREW_CELLAR: /usr/local/Cellar
HOMEBREW_BOTTLE_DOMAIN: https://homebrew.bintray.com
CPU: dual-core 64-bit penryn
Homebrew Ruby: 2.0.0-p648
Clang: 8.0 build 800
Git: 2.10.1 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Perl: /usr/bin/perl
Python: /usr/local/bin/python => /usr/local/Cellar/python/2.7.13/Frameworks/Python.framework/Versions/2.7/bin/python2.7
Ruby: /usr/bin/ruby => /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/bin/ruby
Java: 1.8.0_40
macOS: 10.11.6-x86_64
Xcode: 8.2.1
CLT: 8.2.0.0.1.1480973914
X11: N/A
```

*brew doctor:*
```
Warning: Unbrewed header files were found in /usr/local/include.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected header files:
  /usr/local/include/boost/python/detail/python22_fixed.h
  /usr/local/include/boost/regex.h
  /usr/local/include/boost/tr1/tr1/bcc32/array.h
  /usr/local/include/boost/tr1/tr1/bcc32/random.h
  /usr/local/include/boost/tr1/tr1/bcc32/regex.h
  /usr/local/include/boost/tr1/tr1/bcc32/tuple.h
  /usr/local/include/boost/tr1/tr1/bcc32/type_tra.h
  /usr/local/include/boost/tr1/tr1/bcc32/unordere.h

Warning: You have unlinked kegs in your Cellar
Leaving kegs unlinked can lead to build-trouble and cause brews that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  haskell-platform
```

```
brew gist-logs python
https://gist.github.com/aeabbfb8464129be6cd32b6576e5f73a
```

### Details

I just wanted to run a Django application (I didn't used python for a while):

    [master][~/Development/python/django-projects/return_code]$ python manage.py runserver
    Traceback (most recent call last):
      File "manage.py", line 8, in <module>
        from django.core.management import execute_from_command_line
    ImportError: No module named django.core.management

Hmm, looks like Django is mising.

    [master][~/Development/python/django-projects/return_code]$ pip list
    zsh: command not found: pip

Ohh, pip missing as well. What python is installed?

    [~]$ which -a python
    /usr/local/bin/python
    /usr/bin/python

There is, however, a system pip, but no brew one anymore:

    [~]$ which -a pip-2.7
    /usr/local/bin/pip-2.7

Then I checked easy_install:

    [~]$ which -a easy_install
    /usr/bin/easy_install

Only the system version left.

Then I tried:

    [~]$ brew uninstall --force python
    Uninstalling python... (6,289 files, 85.5M)
    [~]$ brew install python
    ==> Downloading https://homebrew.bintray.com/bottles/python-2.7.13.el_capitan.bo
    Already downloaded: /Users/janwedel/Library/Caches/Homebrew/python-2.7.13.el_capitan.bottle.tar.gz
    ==> Pouring python-2.7.13.el_capitan.bottle.tar.gz
    ==> Using the sandbox
    ==> /usr/local/Cellar/python/2.7.13/bin/python -s setup.py --no-user-cfg install
    Last 15 lines from /Users/janwedel/Library/Logs/Homebrew/python/post_install.01.python:
      File "/usr/local/Cellar/python/2.7.13/libexec/setuptools/setuptools/command/egg_info.py", line 279, in run
        self.find_sources()
      File "/usr/local/Cellar/python/2.7.13/libexec/setuptools/setuptools/command/egg_info.py", line 306, in find_sources
        mm.run()
      File "/usr/local/Cellar/python/2.7.13/libexec/setuptools/setuptools/command/egg_info.py", line 533, in run
        self.add_defaults()
      File "/usr/local/Cellar/python/2.7.13/libexec/setuptools/setuptools/command/egg_info.py", line 572, in add_defaults
        rcfiles = list(walk_revctrl())
      File "/usr/local/Cellar/python/2.7.13/libexec/setuptools/setuptools/command/sdist.py", line 20, in walk_revctrl
        for item in ep.load()(dirname):
      File "/usr/local/Cellar/python/2.7.13/libexec/setuptools/pkg_resources/__init__.py", line 2291, in load
        return self.resolve()
      File "/usr/local/Cellar/python/2.7.13/libexec/setuptools/pkg_resources/__init__.py", line 2297, in resolve
        module = __import__(self.module_name, fromlist=['__name__'], level=0)
    ImportError: No module named setuptools_git
    Warning: The post-install step did not complete successfully
    You can try again using `brew postinstall python`
    ==> Caveats
    Pip and setuptools have been installed. To update them
      pip install --upgrade pip setuptools
    
    You can install Python packages with
      pip install <package>
    
    They will install into the site-package directory
      /usr/local/lib/python2.7/site-packages
    
    See: http://docs.brew.sh/Homebrew-and-Python.html
    ==> Summary
    🍺  /usr/local/Cellar/python/2.7.13: 3,203 files, 43.8M

But still no luck:

    [~]$ pip
    zsh: command not found: pip

The only thing that changed compared to when it was still working AFAIK is the latest El Capitan Update 10.11.6 that I did.

To sum it up: How can I get homebrew pip working again on my Mac?

---

For reference, I already posted this question [here](http://stackoverflow.com/questions/42146594/homebrew-python-2-7-pip-missing-after-el-capitan-update).

**Comment by Numbermaniac** [comment]
Does it work, perhaps, if you try `python -m pip`? Perhaps it's missing from your path.

**Comment by jwedel** [comment]
Unfortunately, not:

```
[~]$ python -m pip
/usr/local/opt/python/bin/python2.7: No module named pip
```

**Comment by ilovezfs** [comment]
Please try
```
mkdir ~/Documents/site-packages-20170305
sudo mv /usr/local/lib/python2.7/site-packages ~/Documents/site-packages-20170305/
brew postinstall python
```

**Comment by jwedel** [comment]
Hi, this is the result:

```
[~]$ mkdir ~/Documents/site-packages-20170305
[~]$ sudo mv /usr/local/lib/python2.7/site-packages ~/Documents/site-packages-20170305/
Password:
[~]$ brew postinstall python
==> Using the sandbox
==> /usr/local/Cellar/python/2.7.13/bin/python -s setup.py --no-user-cfg install
==> /usr/local/Cellar/python/2.7.13/bin/python -s setup.py --no-user-cfg install
==> /usr/local/Cellar/python/2.7.13/bin/python -s setup.py --no-user-cfg install
[~]$ which -a pip
/usr/local/bin/pip
[~]$ pip

Usage:
  pip <command> [options]
(...)
```

Yay!

So what was the problem?

**Comment by ilovezfs** [comment]
Something was corrupt in your site-packages, which persist between installations.

**Comment by ilovezfs** [comment]
Closing since this seems to be resolved. Glad it's working for you now.

**Comment by jwedel** [comment]
Yes, it's resolved!

So brew doctor won't show that issue?

**Comment by ahmedelgabri** [comment]
I have the same exact problem & https://github.com/Homebrew/homebrew-core/issues/10436#issuecomment-284239570 didn't fix it, is there is any different solutions?

**Comment by miaozeyu** [comment]
This fixed it for me!! Thanks!

**Comment by ahmedelgabri** [comment]
@miaozeyu You might want to check this, this turned to be my problem & this was the right solution https://github.com/Homebrew/homebrew-core/issues/15746#issuecomment-316292316
